bark/models/behavior/idm/base_idm.cpp

```diff
@@ -179,15 +179,15 @@ IDMRelativeValues BaseIDM::CalcRelativeValues(
   double leading_acc = 0.0;
   IDMRelativeValues rel_values;
 
-  std::pair<AgentPtr, FrenetPosition> leading_vehicle =
+  auto leading_vehicle =
       observed_world.GetAgentInFront(lane_corr);
 
   // vehicles
   if (leading_vehicle.first) {
-    leading_distance = CalcNetDistance(observed_world, leading_vehicle.first, lane_corr);
+    leading_distance = leading_vehicle.second.lon;
```

Comment:
The FrenetStateDifference already includes the shape extends of the agents and is more accurate since it also consider the orientation of the shapes towards the center lines. This is especially relevant in intersections.